modules/dxp/apps/commerce/commerce-shop-by-diagram/commerce-shop-by-diagram-web/src/main/resources/META-INF/resources/js/AdminTooltip.js

```diff
@@ -92,36 +123,49 @@ const AdminTooltip = ({
 					<label htmlFor={`${namespace}pin-sku`}>
 						{Liferay.Language.get('select-sku')}
 					</label>
-					<ClayAutocomplete>
+
+					<ClayAutocomplete ref={node}>
 						<ClayAutocomplete.Input
-							onChange={(event) => setSku(event.target.value)}
-							value={sku}
-						/>
-						<ClayAutocomplete.DropDown
-							active={skus}
-							onSetActive={(event) => {
-								setSku(
-									event.target.innerText.match(/^[\S]*/g)[0]
-								);
+							onChange={(event) => {
+								setSelectedProduct(null);
+
+								setQuery(event.target.value);
+							}}
+							onFocus={(_e) => {
+								setActive(true);
 							}}
-						>
-							<ClayDropDown.ItemList>
-								{skus?.length && (
-									<ClayDropDown.Item disabled>
-										{Liferay.Language.get(
-											'no-results-found'
-										)}
-									</ClayDropDown.Item>
-								)}
-								{skus?.length &&
-									skus.map((item) => (
-										<ClayAutocomplete.Item
-											key={item.id}
-											match={sku}
-											value={`${item.sku} - ${item.productName.en_US}`}
-										/>
-									))}
-							</ClayDropDown.ItemList>
+							onKeyUp={(event) => {
+								setActive(event.keyCode !== 27);
+							}}
+							value={selectedProduct?.sku || query}
+						/>
+
+						<ClayAutocomplete.DropDown active={active && products}>
+							<div ref={dropdownNode}>
+								<ClayDropDown.ItemList>
+									{products?.length && (
```

Comment:
Anytime we have conditional JSX where we are checking for length, we need to make sure we don't render `0`. If the length is zero here, we will literally render 0.

```suggestion
									{!!products?.length && (
```